### PR TITLE
feature/338_multitenancy

### PR DESCRIPTION
**Pull request by frbattid**
* Implements issue https://github.com/telefonicaid/fiware-cygnus/issues/338
* 100% unit tests passed
```
Results :
Tests run: 49, Failures: 0, Errors: 0, Skipped: 0
```
* (unofficial) e2e tests passed
```
(serviceAsNamespace=true and cosmos_default_username=superuser)

$ curl -X GET "http://130.206.80.46:14000/webhdfs/v1/user/def_serv/def_serv_path/room1_room/room1_room.txt?op=open&user.name=superuser"
{"recvTimeTs":"1430833862","recvTime":"2015-05-05T13:51:02.191Z","entityId":"Room1","entityType":"Room","attrName":"temperature","attrType":"centigrade","attrValue":"26.5","attrMd":[]}

(serviceAsNamespace=false and cosmos_default_username=normaluser)

$ curl -X GET "http://130.206.80.46:14000/webhdfs/v1/user/normaluser/def_serv/def_serv_path/room1_room/room1_room.txt?op=open&user.name=normaluser"
{"recvTimeTs":"1430832905","recvTime":"2015-05-05T13:35:05.88Z","entityId":"Room1","entityType":"Room","attrName":"temperature","attrType":"centigrade","attrValue":"26.5","attrMd":[]}
```
* Assignee @fgalan but LGTM from @gtorodelvalle is also required. Comments from @iariasleon are welcome as well.
* **DO NOT MERGE UNTIL PR https://github.com/telefonicaid/fiware-cygnus/pull/378 IS MERGED INTO DEVELOP.**

### CODE REVIEW

File: doc/design/OrionHDFSSink.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ Assuming the following Flume event is created from a notified NGSI context data
 	    }
     }
 
-Assuming `cosmos_default_username=myuser` and `attr_persistence=row` as configuration parameters, then `OrionHDFSSink` will persist the data within the body as:
+Assuming `cosmos_default_username=myuser`, `service_as_namespace=false` and `attr_persistence=row` as configuration parameters, then `OrionHDFSSink` will persist the data within the body as:
```
Comment (fgalan) [inline doc/design/OrionHDFSSink.md]: This was merged in a previous PR, right? Deja vu... :)

**Comment by gtorodelvalle** [comment]
Not much to say from my side :) In fact, I was just going to suggest extracting the text ```/webhdfs/v1/user/``` in ```HDFSBackendImpl.java``` to a constant to avoid problems in the future when it changes but it is not that important so I leave it to your discretion :)

**Comment by gtorodelvalle** [comment]
LGTM from my side, BTW :)

File: doc/design/OrionHDFSSink.md (hunk 1 above)
Comment (frbattid) [inline doc/design/OrionHDFSSink.md]: Yes... this is the kind of magic I cannot not understand :) Anyway, it is fine.

**Comment by frbattid** [comment]
@gtorodelvalle , I think it is a good idea!

**Comment by fgalan** [comment]
LGTM

**Pull request merged**